public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -102,7 +110,15 @@ const Policy = (props: PolicyComponentProps) => {
 
   const matchers = normalizeMatchers(currentRoute);
   const hasMatchers = Boolean(matchers && matchers.length);
-  const hasFocus = routesMatchingFilters.some((route) => route.id === currentRoute.id);
+
+  const { filtersApplied, matchedRoutes, matchedRoutesWithPath } = routesMatchingFilters;
+
+  // check if this route matches the filters
+  const hasFocus = filtersApplied && matchedRoutes.some((route) => route.id === currentRoute.id);
+
+  // check if this route belongs to a path that matches the filters
+  const routesPath = Array.from(matchedRoutesWithPath.values()).flat();
+  const belongsToMatchPath = routesPath.some((route: RouteWithID) => route.id === currentRoute.id);
 
```

Comment:
I'm a bit concerned that this code can cause performance issues. Each `Policy` component needs to loop over `matchedRoutes` and `matchedRoutesWithPath` . Performance-wise it would be better to, for example build a `Set` object of matching routes Ids and pass it to each `Policy` so it's not recalculated over and over again

Reply:
I tested with 5000 policies and it was pretty fast. Maybe there is a use case I'm missing?

Reply:
I think that if each policy had multiple matchers and if the filter would match most of policies you have